store/src/java/com/zimbra/qa/unittest/TestUtil.java

```diff
@@ -524,19 +525,49 @@ private static ZMimePart getPart(ZMimePart mimeStructure, String name) {
         return null;
     }
 
-    /**
-     * @param numMsgsExpected
-     *            - 0 means don't expect a message to arrive before timeout_millis
-     * @param timeout_millis
-     * @throws ServiceException
-     */
-    public static List<ZMessage> waitForMessages(ZMailbox mbox, String query, int numMsgsExpected, int timeout_millis)
-            throws ServiceException {
-        int orig_timeout_millis = timeout_millis;
-        List<ZMessage> msgs = Lists.newArrayListWithExpectedSize(0);
+    private static void logInputStream(InputStream is, String prefix) {
+        try (InputStreamReader isr = new InputStreamReader(is);
+                BufferedReader ir = new BufferedReader(isr)) {
+            String line;
+            while ((line = ir.readLine()) != null) {
+                ZimbraLog.test.info("mailq STDOUT:%s", line);
+            }
+        } catch (IOException e) {
+            ZimbraLog.test.error("Problem logging stream with prefix '%s'", prefix, e);
+        }
+    }
+
+    private static void execMailq(boolean attemptToDeliverAllQueuedMail) {
```

Comment:
we have a mechanism for executing postfix CLI commands, take a look com.zimbra.cs.rmgmt.RemoteMailQueue and com.zimbra.cs.service.admin.MailQueueAction. I suspect that using that mechanism would be more friendly to testing in multi-node environments.